yarn-project/end-to-end/src/devnet/e2e_devnet.test.ts

```diff
@@ -0,0 +1,226 @@
+import { getSchnorrAccount } from '@aztec/accounts/schnorr';
+import {
+  type EthAddress,
+  Fq,
+  Fr,
+  NativeFeePaymentMethodWithClaim,
+  type PXE,
+  SignerlessWallet,
+  TxStatus,
+  createAztecNodeClient,
+  createPXEClient,
+} from '@aztec/aztec.js';
+import { DefaultMultiCallEntrypoint } from '@aztec/aztec.js/entrypoint';
+import { GasSettings } from '@aztec/circuits.js';
+import { startHttpRpcServer } from '@aztec/foundation/json-rpc/server';
+import { type DebugLogger } from '@aztec/foundation/log';
+import { promiseWithResolvers } from '@aztec/foundation/promise';
+import { GasTokenContract, TestContract } from '@aztec/noir-contracts.js';
+import { createPXERpcServer } from '@aztec/pxe';
+
+import getPort from 'get-port';
+import { exec } from 'node:child_process';
+import { tmpdir } from 'node:os';
+
+import { getLogger, setupPXEService } from '../fixtures/utils.js';
+
+const { AZTEC_NODE_URL, PXE_URL, FAUCET_URL, AZTEC_CLI, ETHEREUM_HOST } = process.env;
+
+describe('End-to-end tests for devnet', () => {
+  // eslint-disable-next-line
+  let pxe: PXE;
+  let pxeUrl: string; // needed for the CLI
+  let logger: DebugLogger;
+  let l1ChainId: number;
+  let feeJuiceL1: EthAddress;
+  let teardown: () => void | Promise<void>;
+
+  beforeAll(async () => {
+    logger = getLogger();
+
+    if (!ETHEREUM_HOST) {
+      throw new Error('ETHEREUM_HOST must be set');
+    }
+
+    if (!AZTEC_CLI) {
+      throw new Error('AZTEC_CLI must be set');
+    }
+
+    if (!FAUCET_URL) {
+      throw new Error('FAUCET_URL must be set');
+    }
+
+    logger.info(`Using AZTEC_CLI: ${AZTEC_CLI}`);
+
+    if (AZTEC_NODE_URL) {
+      logger.info(`Using AZTEC_NODE_URL: ${AZTEC_NODE_URL}`);
+      const node = createAztecNodeClient(AZTEC_NODE_URL);
+      const svc = await setupPXEService(node);
+      pxe = svc.pxe;
+
+      const port = await getPort();
+      pxeUrl = `http://localhost:${port}`;
```

Comment:
I'd use something like the cli's [`getLocalhost`](https://github.com/AztecProtocol/aztec3-packages/blob/master/yarn-project/cli/src/utils/commands.ts#L19-L23) here, so we can run this test in docker envs